### ISSUE TITLE
Is GRPO Adv correctly implemented?

### ISSUE THREAD

**Issue by TebooNok** [PeterGriffinJin/Search-R1#65]
[Adv calculation code in ray_trainer.py#L801](https://github.com/PeterGriffinJin/Search-R1/blob/main/verl/trainer/ppo/ray_trainer.py#L801)

Function compute_advantage has an unused parameter ```num_repeat```, which looks like a sample setting for GRPO.
I think a classic GRPO uses a group sampling from the same state \[s0\], i.e., for one same X generates some different Y, and average them as an estimation for value function.
But in your implementation,
input data is a batch during the training step, Xs in this batch are different, thus averaging them cannot get a critic thus cannot lead to advantage.

```
def compute_advantage(data: DataProto, adv_estimator, gamma=1.0, lam=1.0, num_repeat=1):
    # prepare response group
    # TODO: add other ways to estimate advantages
    if adv_estimator == 'gae':
        values = data.batch['values']
        responses = data.batch['responses']
        response_length = responses.size(-1)
        attention_mask = data.batch['attention_mask']
        response_mask = attention_mask[:, -response_length:]
        token_level_rewards = data.batch['token_level_rewards']
        advantages, returns = core_algos.compute_gae_advantage_return(token_level_rewards=token_level_rewards,
                                                                      values=values,
                                                                      eos_mask=response_mask,
                                                                      gamma=gamma,
                                                                      lam=lam)
        data.batch['advantages'] = advantages
        data.batch['returns'] = returns
    elif adv_estimator == 'grpo':
        token_level_rewards = data.batch['token_level_rewards']
        index = data.non_tensor_batch['uid']
        responses = data.batch['responses']
        response_length = responses.size(-1)
        attention_mask = data.batch['attention_mask']
        response_mask = attention_mask[:, -response_length:]
        advantages, returns = core_algos.compute_grpo_outcome_advantage(token_level_rewards=token_level_rewards,
                                                                        eos_mask=response_mask,
                                                                        index=index)
        data.batch['advantages'] = advantages
        data.batch['returns'] = returns
    else:
        raise NotImplementedError
    return data
```

**Comment by PeterGriffinJin** [comment]
The input X repeat for GRPO is done here: https://github.com/PeterGriffinJin/Search-R1/blob/main/verl/trainer/ppo/ray_trainer.py#L702

The group computation will be based on the index which is repeated: https://github.com/PeterGriffinJin/Search-R1/blob/main/verl/trainer/ppo/ray_trainer.py#L149
